packages/toolkit/src/view/pipeline/view-pipeline/Head.tsx

```diff
@@ -130,242 +130,247 @@ export const Head = ({
-                        Clone
-                      </Button>
-                    }
-                    pipeline={pipeline.data}
-                  />
-                ) : (
-                  <Button
-                    onClick={() => {
-                      navigateBackAfterLogin();
-                    }}
-                    className="!normal-case"
-                    variant="secondaryGrey"
-                    size="sm"
-                  >
-                    Log in to Clone
-                  </Button>
-                )}
-                {pipeline.data.permission.can_edit ? (
-                  <Button
-                    onClick={() => {
-                      router.push(
-                        `/${entity.data.entity}/pipelines/${entity.data.id}/builder`
-                      );
-                    }}
-                    size="sm"
-                    variant="secondaryColour"
-                  >
-                    Build
-                  </Button>
-                ) : null}
-              </React.Fragment>
-            ) : (
-              <HeaderControllerSkeleton />
-            )}
+              {pipeline.isSuccess ? (
+                <React.Fragment>
+                  {pipeline.data.permission.can_edit ? (
+                    <Button
+                      onClick={() => {
+                        router.push(
+                          `/${entity.data.entity}/pipelines/${entity.data.id}/builder`
+                        );
+                      }}
+                      size="sm"
+                      variant="secondaryGrey"
+                      className="gap-x-2"
+                    >
+                      <Icons.Tool01 className="h-3 w-3 stroke-semantic-fg-secondary" />
+                      Edit
+                    </Button>
+                  ) : null}
+                  {me.isSuccess ? (
+                    <ClonePipelineDialog
+                      trigger={
+                        <Button
+                          className="items-center gap-x-2"
+                          size="sm"
+                          variant="secondaryGrey"
+                        >
+                          <Icons.Copy07 className="h-3 w-3 stroke-semantic-fg-secondary" />
+                          Clone
+                        </Button>
+                      }
+                      pipeline={pipeline.data}
+                    />
+                  ) : (
+                    <Button
+                      onClick={() => {
+                        router.push("/login");
+                      }}
+                      className="!normal-case"
+                      variant="secondaryGrey"
+                      size="sm"
+                    >
+                      Log in to Clone
+                    </Button>
+                  )}
+                </React.Fragment>
+              ) : (
+                <HeaderControllerSkeleton />
+              )}
+            </div>
```

Comment:
@iamnamananand996 The whole logic is not correct...

When user can_edit we display edit button and when user is login but can not edit we display clone button. Please double check the original impl